aws-cost-exporter.py

```diff
@@ -6,97 +6,126 @@
-			'End':  end
-		},
-		Granularity='DAILY',
-		Metrics=['BlendedCost'],
-		GroupBy=[
-			{
-				'Type': 'TAG',
-				'Key': tagProject 
-			},
-		]
-	)
-
-	#Create an empty dictionary
-	projectValues = {}
-
-	#Run the response and make a dictionary with tag name and tag value
-	for project in response["ResultsByTime"][0]["Groups"]:
 
-		#Search for tag
-		namestring = project['Keys'][0]
-		name = re.search("\$(.*)", namestring).group(1)
-
-		#If name is none, let's defined it as Other
-		if name is None or name == "":
-			name = "Other"
-
-		#Get the value
-		amount = project['Metrics']['BlendedCost']['Amount']
-		#Format the time to 0.2 points
-		amount = "{0:.2f}".format(float(amount))
-
-		#Append the values in the directionary
-		projectValues[name] = float(amount)
-
-	#Return the dictionary with all those values
-	return projectValues
-
-
-#Start classe collector
+def getCosts():
+    # Create a boto3 connection with cost explorer
+    client = boto3.client('ce')
+
+    # Get the current time
+    now = datetime.datetime.utcnow()
+
+    # Set the end of the range to start of the current day
+    end = datetime.datetime(
+        year=now.year, month=now.month, day=now.day, hour=now.hour)
+
+    # Subtract a day to define the start of the range
+    start =  datetime.datetime(
+        year=now.year, month=now.month, day=now.day, hour=0)
+
+    # Convert them to strings
+    start = start.strftime('%Y-%m-%dT%H:00:00Z')
+    end = end.strftime('%Y-%m-%dT%H:00:00Z')
+
+    print("Starting script searching by the follow time range")
+    print(start + " - " + end)
+
+    # Call AWS API to get costs
+    response = client.get_cost_and_usage(
+        TimePeriod={
+            'Start': start,
+            'End': end
+        },
+        Granularity='HOURLY',
+        Metrics=['BlendedCost'],
+        GroupBy=[
+            {
+                'Type': 'TAG',
+                'Key': 'Product'
+            },
+            {
+                'Type': 'TAG',
+                'Key': 'App'
+            }
+        ]
+    )
+
+    # Create an empty dictionary
+    projectValues = []
+
+    # Run the response and make a dictionary with tag name and tag value
+    for project in response["ResultsByTime"]:
+
+        groups = project["Groups"]
+
+        if not groups or groups is None:
+            continue
+
+        for group in groups:
+            objectValues = {}
+
+            keys = group['Keys']
+            product = keys[0]
+            app = keys[1]
+            product_value = product.split("Product$", 1)[1]
+            app_value = app.split("App$", 1)[1]
+
+            unincoded_product_value = ast.literal_eval(json.dumps(product_value))
+            unincoded_app_value = ast.literal_eval(json.dumps(app_value))
+          
+
+            if (not unincoded_product_value or unincoded_product_value is None or unincoded_product_value == '') and (not unincoded_app_value or unincoded_app_value is None or unincoded_app_value == ''):
```

Comment:
I'm pretty sure you don't need this complex conditiona, since python returns `True` for `not unincoded_product_value` if it's `None` or  a empty string (`''`), example:
```python
>>> unincoded_product_value = None
>>> not unincoded_product_value
True
>>> unincoded_product_value = ''
>>> not unincoded_product_value
True
```

a simple `if (not unincoded_product_value) and (not unincoded_app_value):` will do the work.

Reply:
great! I will ref this! thanks!!